src/test/java/com/example/java/maven/ticketVendingMachine/CoinStorageTest.java

```diff
@@ -0,0 +1,86 @@
+package com.example.java.maven.ticketVendingMachine;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.mockito.BDDMockito.then;
+
+import java.util.Arrays;
+import java.util.HashMap;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.times;
+
+
+public class CoinStorageTest {
+    MessagePrinter messagePrinter = Mockito.mock(MessagePrinter.class);
+
+    CoinStorage primaryCoinStorage;
+    CoinStorage destinationCoinStorage;
+
+
+    @BeforeEach
+    void createCoinStorages() {
+        this.primaryCoinStorage = new CoinStorage(new HashMap<>(), messagePrinter);
+        this.destinationCoinStorage = new CoinStorage(new HashMap<>(), messagePrinter);
+    }
+
+
+    @Test
+    void shouldAddCoin() {
+        Coin coin = Coin.ONEZL;
+
+//        when
+        primaryCoinStorage.addCoin(coin);
+
+//        then
+        assertThat(primaryCoinStorage.getCoins().size()).isEqualTo(1);
+    }
+
+    @Test
+    void shouldFillCoinStorageWithGivenNumberOfCoinsForDenomination() {
+
+//        when
+        primaryCoinStorage.fillCoinStorageWithGivenNumberOfCoinsForDenomination(1);
+
+//        then
+        assertThat(primaryCoinStorage.getCoins().size()).isEqualTo(6);
```

Comment:
nice - but I needed a moment to understand why six.
It is good practice to add comment with explanation in this case, so for example:
```java
assertThat(primaryCoinStorage.getCoins().size()).isEqualTo(6); \\ 6 - because we have 6 types of Coins - so we expect single monet of every type in storage
```

Reply:
Roger that